internal/testutil/kube_api_http_client.go

```
@@ -0,0 +1,67 @@
+package testutil
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func GetEndpointOnCluster(ctx context.Context, namespace, service, path string, port int) ([]byte, error) {
+	restConfig := config.GetConfigOrDie()
+	transport, err := rest.TransportFor(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct an http.Client that authenticates with the apiserver.
+	client := &http.Client{
+		Transport: transport,
+	}
+
+	// Construct request
+	requestURL, err := createURL(restConfig.Host, namespace, service, path, port)
+	if err != nil {
+		return nil, err
+	}
+
+	req, _ := http.NewRequestWithContext(ctx, http.MethodPost, requestURL.String(), nil) // OK
```

should be a GET

---

don't discard error + handle it